### PR TITLE
P#50191 User-friendly explanation of PDF brochure indexing

### PR DESCRIPTION
**Pull request by vu-vuong-esg**

### CODE REVIEW

File: plugin/Model/InputModelBase.php [hunk 1]
```diff
@@ -102,6 +102,9 @@ abstract class InputModelBase
 	/** @var array */
 	private $_labelOnlyValues = [];
 
+	/** @var string */
+	private $_descriptionTextHTML = null;
```
Comment (jayay) [inline plugin/Model/InputModelBase.php:106]: Better initialize as empty string, then add the `string` return type to the getter

File: plugin/Renderer/InputFieldCheckboxRenderer.php [hunk 2]
```diff
@@ -124,7 +127,8 @@ public function render()
 				.'" value="'.esc_html($this->getValue()).'"'
 				.($this->getValue() == $this->_checkedValues ? ' checked="checked" ' : '')
 				.$this->renderAdditionalAttributes()
-				.' id="'.esc_html($this->getGuiId()).'">';
+				.' id="'.esc_html($this->getGuiId()).'">'
+				.(!empty($this->_descriptionTextHTML) && is_string($this->_descriptionTextHTML) ? '<p class="description">'.$this->_descriptionTextHTML.'</p><br>' : '');
```
Comment (jayay) [inline plugin/Renderer/InputFieldCheckboxRenderer.php:131]: better check for `!== ''` instead of `is_string()`

**Review by jayay — CHANGES_REQUESTED** [review]
One small thing, but mostly good!

File: plugin/Model/InputModelBase.php (hunk 1 above)
Comment (vu-vuong-esg) [reply plugin/Model/InputModelBase.php:106]: ok I will update

File: plugin/Renderer/InputFieldCheckboxRenderer.php (hunk 2 above)
Comment (vu-vuong-esg) [reply plugin/Renderer/InputFieldCheckboxRenderer.php:131]: ok I will update

**Review by jayay — APPROVED** [review]

**Comment by michaelarnold-dev** [comment]
@hungnc89 Could you please resolve the conflicts in this branch?

**Comment by tung-le-esg** [comment]
> @tung-le-esg Could you please resolve the conflicts in this branch?

Okay. I will resolve the conflict files

**Comment by michaelarnold-dev** [comment]
@tung-le-esg I guess you are still analyzing the conflicts, right?

**Comment by tung-le-esg** [comment]
> @tung-le-esg I guess you are still analyzing the conflicts, right?

@michaelarnold-dev  This PR is in the queue. I will be back to this PR soon.

**Comment by tung-le-esg** [comment]
@michaelarnold-dev Conflicts have been resolved.

File: plugin/Model/InputModelBase.php [hunk 3]
```diff
@@ -108,8 +108,8 @@ abstract class InputModelBase
 	/** @var array */
 	private $_labelOnlyValues = [];
 
-	/** @var string */
-	private $_descriptionTextHTML = '';
+    /** @var string */
+    private $_descriptionTextHTML = [];
```
Comment (jayay) [inline plugin/Model/InputModelBase.php]: Why init this with an array?

**Review by jayay — CHANGES_REQUESTED** [review]
I don't understand what the last changes are supposed to do. I think the last changes should be reverted. If this wasn't intentional, I would kindly ask you for a fix

**Comment by michaelarnold-dev** [comment]
@tung-le-esg : Did you see the comment of Jacob?

**Comment by tung-le-esg** [comment]
> @tung-le-esg : Did you see the comment of Jacob?

Yes. I have already committed the fix.

**Review by jayay — APPROVED** [review]

**Pull request merged**